src/main/java/io/lettuce/core/ReadFromImpl.java

```diff
@@ -129,6 +136,60 @@ public ReadFromAnyReplica() {
 
     }
 
+    /**
+     * Read from any node in the subnets.
+     *
+     * @since x.x.x
+     */
+    static final class ReadFromSubnet extends ReadFrom {
+
+        private final List<IpSubnetFilterRule> rules = new ArrayList<>();
+
+        /**
+         * @param cidrNotations CIDR-block notation strings, e.g., "192.168.0.0/16".
+         */
+        ReadFromSubnet(String... cidrNotations) {
+            LettuceAssert.notEmpty(cidrNotations, "cidrNotations must not be empty");
+
+            for (String cidrNotation : cidrNotations) {
+                // parts[0]: ipAddress (e.g., "192.168.0.0")
+                // parts[1]: cidrPrefix (e.g., "16")
+                String[] parts = cidrNotation.split("/");
+                LettuceAssert.isTrue(parts.length == 2, "cidrNotation must have exact one '/'");
+
+                rules.add(new IpSubnetFilterRule(parts[0], Integer.parseInt(parts[1]), IpFilterRuleType.ACCEPT));
+            }
+        }
+
+        @Override
+        public List<RedisNodeDescription> select(Nodes nodes) {
+            List<RedisNodeDescription> result = new ArrayList<>(nodes.getNodes().size());
+            for (RedisNodeDescription node : nodes) {
+                if (isInSubnet(node.getUri())) {
+                    result.add(node);
+                }
+            }
+
+            return result;
+        }
+
+        private boolean isInSubnet(RedisURI redisURI) {
+            if (LettuceStrings.isEmpty(redisURI.getHost())) {
+                return false;
+            }
+
+            InetSocketAddress address = SocketUtils.socketAddress(redisURI.getHost(), redisURI.getPort());
```

Comment:
This line can lead to a DNS lookup (blocking call). Would `createUnresolved` work as well?